### PR TITLE
Update BesuController to work with NodeKey (not KeyPair)

### PR DESCRIPTION
**Pull request by rain-on**
## PR description
Update BesuController to use the NodeKey, rather than working with KeyPair - which in turn allows the crypto operations to be injected.

## Fixed Issue(s)
N/A

### CODE REVIEW

File: acceptance-tests/dsl/src/main/java/org/hyperledger/besu/tests/acceptance/dsl/node/ThreadBesuNodeRunner.java [hunk 1]
```diff
@@ -134,26 +133,21 @@ public void startNode(final BesuNode node) {
...
-    } catch (final IOException e) {
-      throw new RuntimeException("Error building BesuController", e);
-    }
+    final BesuController<?> besuController =
+        builder
+            .synchronizerConfiguration(new SynchronizerConfiguration.Builder().build())
+            .dataDirectory(node.homeDirectory())
+            .miningParameters(node.getMiningParameters())
+            .privacyParameters(node.getPrivacyParameters())
+            .nodePrivateKeyFile(KeyPairUtil.getDefaultKeyFile(node.homeDirectory()))
```
Comment (usmansaleem) [inline acceptance-tests/dsl/src/main/java/org/hyperledger/besu/tests/acceptance/dsl/node/ThreadBesuNodeRunner.java:142]: Is there any change here except removing try/catch block?

File: besu/src/main/java/org/hyperledger/besu/controller/BesuControllerBuilder.java [hunk 2]
```diff
@@ -85,7 +85,7 @@
   protected PrivacyParameters privacyParameters;
   protected Path dataDirectory;
   protected Clock clock;
-  KeyPair nodeKeys;
+  NodeKey nodeKey;
```
Comment (usmansaleem) [inline besu/src/main/java/org/hyperledger/besu/controller/BesuControllerBuilder.java:88]: shouldn't this be `protected` or `private`?

File: besu/src/main/java/org/hyperledger/besu/controller/CliqueBesuControllerBuilder.java [hunk 3]
```diff
@@ -57,7 +56,7 @@
 
   @Override
   protected void prepForBuild() {
-    localAddress = Util.publicKeyToAddress(nodeKeys.getPublicKey());
+    localAddress = Util.publicKeyToAddress(nodeKey.getPublicKey());
```
Comment (usmansaleem) [inline besu/src/main/java/org/hyperledger/besu/controller/CliqueBesuControllerBuilder.java:59]: nit: `getNodeKey` instead of accessing super class member variable.

File: besu/src/main/java/org/hyperledger/besu/controller/IbftBesuControllerBuilder.java [hunk 4]
```diff
@@ -208,7 +204,6 @@ protected MiningCoordinator createMiningCoordinator(
 
   @Override
   protected PluginServiceFactory createAdditionalPluginServices(final Blockchain blockchain) {
-    final NodeKey nodeKey = new BouncyCastleNodeKey(nodeKeys);
     return new IbftQueryPluginServiceFactory(blockchain, nodeKey);
```
Comment (usmansaleem) [inline besu/src/main/java/org/hyperledger/besu/controller/IbftBesuControllerBuilder.java:207]: same as before, getNodeKey() instead of nodeKey??

File: acceptance-tests/dsl/src/main/java/org/hyperledger/besu/tests/acceptance/dsl/node/ThreadBesuNodeRunner.java (hunk 1 above)
Comment (rain-on) [reply acceptance-tests/dsl/src/main/java/org/hyperledger/besu/tests/acceptance/dsl/node/ThreadBesuNodeRunner.java:142]: not sure why this is no longer required - will look into it - but yes, its the removal of the try/catch

File: besu/src/main/java/org/hyperledger/besu/controller/BesuControllerBuilder.java (hunk 2 above)
Comment (rain-on) [reply besu/src/main/java/org/hyperledger/besu/controller/BesuControllerBuilder.java:88]: no - this is a builder, so this field has a setter.

File: besu/src/main/java/org/hyperledger/besu/controller/CliqueBesuControllerBuilder.java (hunk 3 above)
Comment (rain-on) [reply besu/src/main/java/org/hyperledger/besu/controller/CliqueBesuControllerBuilder.java:59]: trying to leave as much untouched as possible - so wanted to leave this as is

Comment (usmansaleem) [reply besu/src/main/java/org/hyperledger/besu/controller/CliqueBesuControllerBuilder.java:59]: fair enough.

File: besu/src/main/java/org/hyperledger/besu/controller/IbftBesuControllerBuilder.java (hunk 4 above)
Comment (rain-on) [reply besu/src/main/java/org/hyperledger/besu/controller/IbftBesuControllerBuilder.java:207]: as above - trying to leave things untouched.

File: besu/src/main/java/org/hyperledger/besu/cli/BesuCommand.java [hunk 5]
```diff
@@ -1224,40 +1224,36 @@ private BesuCommand controller() {
...
-          .isPruningEnabled(isPruningEnabled())
-          .pruningConfiguration(
-              new PrunerConfiguration(pruningBlockConfirmations, pruningBlocksRetained))
-          .genesisConfigOverrides(genesisConfigOverrides)
-          .targetGasLimit(targetGasLimit == null ? Optional.empty() : Optional.of(targetGasLimit))
-          .requiredBlocks(requiredBlocks);
-    } catch (final IOException e) {
-      throw new ExecutionException(this.commandLine, "Invalid path", e);
-    }
+    addConfigurationService();
```
Comment (jframe) [inline besu/src/main/java/org/hyperledger/besu/cli/BesuCommand.java:1227]: Why did you remove the try/catch here?

File: besu/src/main/java/org/hyperledger/besu/controller/BesuControllerBuilder.java (hunk 2 above)
Comment (jframe) [reply besu/src/main/java/org/hyperledger/besu/controller/BesuControllerBuilder.java:88]: Looks like it could be protected

File: ethereum/p2p/src/main/java/org/hyperledger/besu/ethereum/p2p/network/DefaultP2PNetwork.java [hunk 6]
```diff
@@ -431,7 +429,7 @@ private P2PNetwork doBuild() {
     }
 
     private void validate() {
-      checkState(keyPair != null, "KeyPair must be set.");
+      checkState(nodeKey != null, "nodeKey must be set.");
```
Comment (jframe) [inline ethereum/p2p/src/main/java/org/hyperledger/besu/ethereum/p2p/network/DefaultP2PNetwork.java:432]: nit: NodeKey instead of nodeKey

File: ethereum/p2p/src/main/java/org/hyperledger/besu/ethereum/p2p/rlpx/RlpxAgent.java [hunk 7]
```diff
@@ -563,17 +563,17 @@ public RlpxAgent build() {
     }
 
     private void validate() {
-      checkState(keyPair != null, "KeyPair must be configured");
+      checkState(nodeKey != null, "nodeKey must be configured");
```
Comment (jframe) [inline ethereum/p2p/src/main/java/org/hyperledger/besu/ethereum/p2p/rlpx/RlpxAgent.java:566]: nit: nodeKey instead of NodeKey

File: besu/src/main/java/org/hyperledger/besu/cli/BesuCommand.java (hunk 5 above)
Comment (rain-on) [reply besu/src/main/java/org/hyperledger/besu/cli/BesuCommand.java:1227]: The IOException is caught in the loader, then rethrown as an IllegalArgumentException - so this seemed superfluous.

File: ethereum/p2p/src/main/java/org/hyperledger/besu/ethereum/p2p/network/DefaultP2PNetwork.java (hunk 6 above)
Comment (rain-on) [reply ethereum/p2p/src/main/java/org/hyperledger/besu/ethereum/p2p/network/DefaultP2PNetwork.java:432]: done

File: ethereum/p2p/src/main/java/org/hyperledger/besu/ethereum/p2p/rlpx/RlpxAgent.java (hunk 7 above)
Comment (rain-on) [reply ethereum/p2p/src/main/java/org/hyperledger/besu/ethereum/p2p/rlpx/RlpxAgent.java:566]: done
